### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by bizip**
**What does this PR do?**

- Install CanCanCan in your project.

- Add a role column to the user's table. Remember to use migration for this.

- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

- For that, you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

- For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

**Description of Task to be completed**

All tasks are completed 

**How should this be manually tested?**

1. Clone this repository and switch to `feature-authorization`

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -17,6 +17,8 @@
                 <h2>Post #<%= post.id%></h1>
                 <div class="post_title">
                    <h2><%= link_to post.title, user_post_path(post.author, post) %></h2>
+                   <button type="button" class="btn btn-warning">Edit</button>
+                  <button type="button" class="btn btn-danger">Delete</button>
                     <p><%= post.text %></p>
                      <div class="post_summary">
                         <p>Comments: <%= post.comments_count.nil? ? 0 : post.comments_count %>  </p>
```
Comment (KDlamini) [inline app/views/posts/index.html.erb:28]: I cannot find a button for Creating a new post on your page. Kindly make sure that there's a button for creating a new post for the current user (logged-in user).

image

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @bizip, @tugsem 🙋‍♀️,  

Excellent work! 💯 
There's always room for improvement. There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 
✔️  CanCanCan gem with the Ability class is used for authorization
✔️  Good PR title and description
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -11,12 +11,18 @@
             
 
 <ul>
+
+  <% if @user == current_user %>
+    <%= link_to 'Add a new post', new_user_post_path %>
```
Comment (addod19) [inline app/views/posts/index.html.erb:16]: - [ ] I'm unable to find the button to add new post to the UI.. Here are a few screenshots of the application

Screenshot from 2022-09-02 16-03-41

Screenshot from 2022-09-02 15-59-51

**Review by addod19 — CHANGES_REQUESTED** [review]
H @bizip and @tugsem, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] UI looks great :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [ ] It would be better if we use cancancan for post authorization in Post controller. For example;
```
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end

```

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @bizip 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**